### PR TITLE
move j2 template to erb in 7.17

### PR DESCRIPTION
**Pull request by jsvd**
This should simplify CI infrastructure by not requiring python.

Exhaustive test suite with docker acceptance tests passing here: https://buildkite.com/elastic/logstash-exhaustive-tests-pipeline/builds/576

### CODE REVIEW

File: docker/templates/Dockerfile.erb [hunk 1]
```diff
@@ -1,37 +1,33 @@
...
+<%   package_manager = 'yum' -%>
+<% else -%>
+<%   base_image = 'ubuntu:20.04' -%>
+<%   package_manager = 'apt-get' -%>
+<%   locale = 'en_US.UTF-8' -%>
+<% end -%>
+<% if image_flavor == 'ironbank' -%>
 ARG BASE_REGISTRY=registry1.dsop.io
-ARG BASE_IMAGE=ironbank/redhat/ubi/ubi8
+ARG BASE_IMAGE=ironbank/redhat/ubi/ubi9
```
Comment (jsvd) [inline docker/templates/Dockerfile.erb:28]:
```suggestion
ARG BASE_IMAGE=ironbank/redhat/ubi/ubi8
```

File: docker/templates/hardening_manifest.yaml.erb [hunk 2]
```diff
@@ -39,27 +39,14 @@ labels:
...
-  url: https://dl.google.com/go/go1.17.8.linux-amd64.tar.gz
-  validation:
-    type: sha256
-    value: 980e65a863377e69fd9b67df9d8395fd8e93858e7a24c9f55803421e453f4f99
-- filename: v2.3.0.tar.gz
-  url: https://github.com/go-yaml/yaml/archive/v2.3.0.tar.gz
-  validation:
-    type: sha512
-    value: ba934e9cb5ebd2346d3897308b71d13bc6471a8dbc0dc0d46a02644ee6b6553d20c20393471b81025b572a9b03e3326bde9c3e8be156474f1a1f91ff027b6a4f
+    value: <INSERT SHA512 VALUE FROM https://artifacts.elastic.co/downloads/logstash/logstash-<%= elastic_version %>-linux-x86_64.tar.gz.sha512>
```
Comment (jsvd) [inline docker/templates/hardening_manifest.yaml.erb:46]:
```suggestion
- filename: go1.17.8.linux-amd64.tar.gz
  url: https://dl.google.com/go/go1.17.8.linux-amd64.tar.gz
  validation:
    type: sha256
    value: 980e65a863377e69fd9b67df9d8395fd8e93858e7a24c9f55803421e453f4f99
- filename: v2.3.0.tar.gz
  url: https://github.com/go-yaml/yaml/archive/v2.3.0.tar.gz
  validation:
    type: sha512
    value: ba934e9cb5ebd2346d3897308b71d13bc6471a8dbc0dc0d46a02644ee6b6553d20c20393471b81025b572a9b03e3326bde9c3e8be156474f1a1f91ff027b6a4f
    value: <INSERT SHA512 VALUE FROM https://artifacts.elastic.co/downloads/logstash/logstash-<%= elastic_version %>-linux-x86_64.tar.gz.sha512>
```

File: docker/templates/hardening_manifest.yaml.erb [hunk 3]
```diff
@@ -39,11 +39,10 @@ labels:
 
 # List of resources to make available to the offline build context
 resources:
-- filename: logstash-{{ elastic_version }}-linux-x86_64.tar.gz
-  url: <artifact_path>/logstash-{{ elastic_version }}-linux-x86_64.tar.gz
+- filename: logstash-<%= elastic_version %>-linux-x86_64.tar.gz
+  url: <artifact_path>/logstash-<%= elastic_version %>-linux-x86_64.tar.gz
   validation:
     type: sha512
```
Comment (jsvd) [inline docker/templates/hardening_manifest.yaml.erb:45]:
```suggestion
    type: sha512
    value: <INSERT SHA512 VALUE FROM https://artifacts.elastic.co/downloads/logstash/logstash-<%= elastic_version %>-linux-x86_64.tar.gz.sha512>

```

File: docker/templates/hardening_manifest.yaml.erb [hunk 4]
```diff
@@ -54,12 +55,10 @@ resources:
...
     value: ba934e9cb5ebd2346d3897308b71d13bc6471a8dbc0dc0d46a02644ee6b6553d20c20393471b81025b572a9b03e3326bde9c3e8be156474f1a1f91ff027b6a4f
+    value: <INSERT SHA512 VALUE FROM https://artifacts.elastic.co/downloads/logstash/logstash-<%= elastic_version %>-linux-x86_64.tar.gz.sha512>
```
Comment (jsvd) [inline docker/templates/hardening_manifest.yaml.erb:58]:
```suggestion
```

**Review by dliappis — APPROVED** [review]
Thank you!

Successfully tested with test images created without Python 3.6 installed: GCP: https://buildkite.com/elastic/logstash-dra-snapshot-pipeline/builds/1113 / AWS: https://buildkite.com/elastic/logstash-dra-snapshot-pipeline/builds/1114

LGTM!

**Pull request merged**